drivers/bmx280/bmx280.c

```diff
@@ -23,93 +25,322 @@
 #include <math.h>
 
 #include "log.h"
+#include "assert.h"
 #include "bmx280.h"
 #include "bmx280_internals.h"
-#include "bmx280_params.h"
-#include "periph/i2c.h"
 #include "xtimer.h"
 
 #define ENABLE_DEBUG        (0)
 #include "debug.h"
 
-static int read_calibration_data(bmx280_t* dev);
-static int do_measurement(const bmx280_t* dev);
-static uint8_t get_ctrl_meas(const bmx280_t* dev);
-static uint8_t get_status(const bmx280_t* dev);
-static uint8_t read_u8_reg(const bmx280_t* dev, uint8_t reg);
-static void write_u8_reg(const bmx280_t* dev, uint8_t reg, uint8_t b);
-static uint16_t get_uint16_le(const uint8_t *buffer, size_t offset);
-static int16_t get_int16_le(const uint8_t *buffer, size_t offset);
-
-#if ENABLE_DEBUG
-static void dump_buffer(const char *txt, uint8_t *buffer, size_t size);
-#define DUMP_BUFFER(txt, buffer, size)  dump_buffer(txt, buffer, size)
+#ifdef BMX280_USE_SPI
+#define BUS                 (dev->params.spi)
+#define CS                  (dev->params.cs)
+#define CLK                 (dev->params.clk)
+#define MODE                SPI_MODE_0
+#define WRITE_MASK          (0x7F)
 #else
-#define DUMP_BUFFER(txt, buffer, size)
+#define BUS                 (dev->params.i2c_dev)
+#define ADDR                (dev->params.i2c_addr)
 #endif
 
+/* shortcut for accessing byte x of the latest sensor reading */
+#define RAW_DATA            (dev->last_reading)
+
+/* implementation for the driver's configured bus interface (I2C vs SPI) */
+#ifdef BMX280_USE_SPI /* using SPI mode */
+static inline int _acquire(const bmx280_t *dev)
+{
+    if (spi_acquire(BUS, CS, MODE, CLK) != SPI_OK) {
+        return BMX280_ERR_BUS;
+    }
+    return BMX280_OK;
+}
+
+static inline void _release(const bmx280_t *dev)
+{
+    spi_release(BUS);
+}
+
+static int _read_reg(const bmx280_t *dev, uint8_t reg, uint8_t *data)
+{
+    *data = spi_transfer_reg(BUS, CS, reg, 0);
+    return BMX280_OK;
+}
+
+static int _write_reg(const bmx280_t *dev, uint8_t reg, uint8_t data)
+{
+    (void)spi_transfer_reg(BUS, CS, (reg & WRITE_MASK), data);
+    return BMX280_OK;
+}
+
+static int _read_burst(const bmx280_t *dev, uint8_t reg, void *buf, size_t len)
+{
+    spi_transfer_regs(BUS, CS, reg, NULL, buf, len);
+    return BMX280_OK;
+}
+
+#else /* using I2C mode */
+
+static inline int _acquire(const bmx280_t *dev)
+{
+    if (i2c_acquire(BUS) != 0) {
+        return BMX280_ERR_BUS;
+    }
+    return BMX280_OK;
+}
+
+static inline void _release(const bmx280_t *dev)
+{
+    i2c_release(BUS);
+}
+
+static int _read_reg(const bmx280_t *dev, uint8_t reg, uint8_t *data)
+{
+    if (i2c_read_reg(BUS, ADDR, reg, &data, 0) != 1) {
```

Comment:
I tested the bme280 driver. An I got this errormessage: [bmx280] error: unable to read chip ID from device 
So you you have to change the _read_reg method. data is already a pointer and  the condition have to be 0.